taskiq/result_backends/redis/redis_client.py

```diff
@@ -0,0 +1,72 @@
+from typing import Mapping, Optional, Union
+
+from errors import RedisConnectionError
+from redis.asyncio import ConnectionPool, Redis
+
+
+class RedisClient:
+    """Class to interact with redis."""
+
+    def __init__(self, redis_url: str) -> None:
+        """
+        Initializes redis connection.
+
+        :param redis_url: Redis URL to connect.
+        """
+        self.redis_pool = ConnectionPool.from_url(redis_url)
+
+    async def close(self) -> None:
+        """Closes redis connection."""
+        await self.redis_pool.disconnect()
+
+    async def hset(
+        self,
+        name: str,
+        mapping: Mapping[Union[str, bytes], Union[bytes, float, int, str]],
+    ) -> None:
+        """
+        Adds new key-value in a redis hashmap with name `name`.
+
+        :param name: The name of the redis hashmap.
+        :param mapping: Dictionary with key-value pairs.
+
+        :raises RedisConnectionError: if redis is not available.
+        """
+        try:
+            async with Redis(connection_pool=self.redis_pool) as redis:
+                await redis.hset(name, mapping=mapping)
+        except ConnectionError as exc:
+            raise RedisConnectionError("Redis is unavailable") from exc
```

Comment:
You don't have to add exception handlers here. 

I guess we will add global `try: catch` in taskiq itself. So it will throw TaskiqErrors if something bad happens during setting or getting the result.

Reply:
https://github.com/taskiq-python/taskiq/issues/13